### ISSUE TITLE
Support LVM

### ISSUE THREAD

**Issue by mdhorn** [clearlinux/clr-installer#8]
<a href="https://github.com/mdhorn"><img src="https://avatars.githubusercontent.com/mdhorn" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [mdhorn](https://github.com/mdhorn)**
_Friday Mar 02, 2018 at 22:18 GMT_

----

Future Enhancements: LVM, Resizing partitions

**Comment by mdhorn** [comment]
Note: We will need to add support for LVM to the special initd similar to what was done to support Encryption for the root file systems:

**initrd=\EFI\org.clearlinux\initrd-org.clearlinux.native.4.20.8-696 initrd=\EFI\org.clearlinux\freestanding-clr-init.cpio.gz** root=UUID=919a625a-8e45-4b08-9e97-e4e6bf4621cb **rd.luks.uuid=6cd79a0f-ed3e-47c5-9e81-016e9683e2fe** quiet console=tty0 console=ttyS0,115200n8 init=/usr/bin/initra-desktop  initcall_debug tsc=reliable no_timer_check noreplace-smp kvm-intel.nested=1 rootfstype=ext4,btrfs,xfs intel_iommu=igfx_off cryptomgr.notests rcupdate.rcu_expedited=1 rcu_nocbs=0-64 rw **rootflags=x-systemd.device-timeout=0**

**Comment by NamPNQ** [comment]
Can you tell me how to use lvm?

**Comment by mdhorn** [comment]
When LVM support is enabled for the Installer, we need to ensure we can overwrite/reuse existing disks which already have LVM enabled.

Currently, if we attempt to overwrite a disk enabled with LVM, it will fail:
Steps to Repro:

1. Install CentOS and choose automatic partition options
2. Start Clear installer
3. Choose "Destructive Installation"
4. Proceed to install and it will fail

 To clear the LVMs prior to installing, these steps seem to work:
```
$ sudo sgdisk -Z /dev/<device>
$ sudo partprobe
$ sudo dmsetup remove_all --force
$ sudo partprobe

```

**Comment by mdhorn** [comment]
Clear Linux OS requirement to enable LVM
https://github.com/clearlinux/distribution/issues/912

**Comment by mdhorn** [comment]
Ensure we have software RAID support also while implementing LVM support.

**Comment by mdhorn** [comment]
Blocked by #530

**Comment by mdhorn** [comment]
Fixed in Clear Linux Installer release 2.7.0 first released in Clear Linux OS 33450.
